### ISSUE TITLE
Web mapping

### ISSUE THREAD

**Issue by sabas** [italia-it/designer.italia.it#55]
Bisognerebbe considerare l'inserimento di una sezione con linee guida sull'inserimento di mappe web all'interno dei siti della PA. Questo poiché è sempre più diffusa la necessità di produrre contesto geografico nella presentazione dei dati, anche solo nelle pagine di 'anagrafica' degli uffici.
Visto anche l'utilizzo di Leaflet su governo.it, suggerirei di costruire una specifica attorno a Leaflet (eventualmente lasciando spazio a OpenLayers3 per cose più avanzate?).
Come base layer chiederei se possibile di utilizzare requisiti abbastanza forti che spingano verso l'open (OpenStreetMap + ortofoto tipo RealVista o PCN, licenziate CC-BY)

**Comment by gunzip** [comment]
ok, ma non la metterei come prioritaria per l'uscita della  versione beta. 

puoi condividere una bozza ?

**Comment by sabas** [comment]
Su che pagina mi posso basare?

**Comment by gunzip** [comment]
@sabas non ho esempi, anche un indice di massima andrebbe bene.

**Comment by napo** [comment]
Provo a buttare giù una bozza tutta da discutere e rivedere

# Uso di mappe nei siti della PA
## Introduzione
È sempre più diffusa la necessità di produrre contesto geografico nella presentazione dei dati, anche solo nelle pagine di 'anagrafica' degli uffici.
Le mappe sono rappresentazioni del territorio e, i loro contenuti, potrebbero non essere neutrali, inoltre è importante valorizzare anche il lavoro dei sistemi informativi territoriali delle varie PA a partire da quelle nazionali.
Questa sezione ha lo scopo di sensibilizzare sulla scelta della mappa da utilizzare come sfondo.
Sulle questioni relative alla neutralità dell'informazione geografica si rimanda a questo articolo
http://blog.spaziogis.it/2014/01/10/perche-abbiamo-bisogno-di-openstreetmap/

## La scelta della mappa di background
I moderni browser, grazie al miglioramento sempre crescente degli interpreti javascript e delle librerie che permettono la visualizzazione di geodati online, sono in grado di gestire con una buona facilità elenchi di punti (o linee o geometrie) da rappresentare.
Questo implica pertanto che, il dato che si vuole mostrare (es. i luoghi degli uffici pubblici), sia facilmente rappresentabile in "overlay" su una mappa.
Le scelte tecniche spesso ricadono su strumenti come Google Maps senza porsi alcuna questione invece sul vero significato del riuso di queste mappe.
La proposta di queste linee guida è quella di selezionare mappe, prima di tutto, generate dai dati raccolti da una pubblica amministrazione. È indubbio che la toponomastica è stabilita dai Comuni e non da i produttori di mappe online.
Di contro però, non tutte le pubbliche amministrazioni sono in grado di ridistribuire i propri dati secondo questi sistemi informativi.
Si propone pertanto di seguire il seguente schema nella scelta della cartografia di sfondo:
1. fare uso di cartografia prodotta dalla PA di riferimento
qualora l'ente ne distribuisca una sua si consiglia vivamente di fare uso della propria mappa in background, meglio se questa aggiornata anche tempestivamente con l'aggiornamento dei dati raccolti.
2. fare uso di cartografia prodotta da un ente centrale
tutte le regioni e province autonome dispongono di servizi SIT che offrono servizi di distribuzione della cartografia di base (carta tecnica o ortofotocarta) basati su protocolli standard (si veda la specifica INSPIRE).
Lo stesso portale cartografico nazionale offre diverse sorgenti in tal senso.
È importante che questa cartografia sia disponibile con licenze aperte al fine di evitare possibile confusione da parte dell'utente finale.
3. fare uso di OpenStreetMap
Il progetto OpenStreetMap offre una banca dati di dati georiferiti a cui chiunque può contribuire.
I dati raccolti, oltre ad essere messi a disposizione di tutti, sono da subito trasformati in mappe online attraverso diversi stili di rappresentazione.
Queste mappe spesso hanno al loro interno una ricchezza di dettagli superiori, garantiscono neutralità dei contenuti e possono essere facilmente aggiornate.

## note tecniche su come le mappe devono essere distribuite
### protocolli e sistemi di riferimento
- uso di TMS o WMS
- uso di WGS84 (EPSG:4326) o Google Mercator (EPSG:3857)
### sorgenti disponibili
#### risorse nazionali
elencare anche l'uso foto aeree georiferite (ortofotocarta)
- come si usano
- esempi
#### risorse regionali o comunali
elencare anche l'uso foto aeree georiferite (ortofotocarta)
- come si usano
- esempi
#### openstreetmap
- come si usa
- esempi
- vari rendering disponibili

## librerie javascript disponibili
## L'esempio del governo inglese con l'Ordnance Survey
https://www.ordnancesurvey.co.uk/business-and-government/products/opendata-products-grid.html

**Comment by sabas** [comment]
Possiamo dividere in più sezioni:
- [ ] Principi
- [ ] Indicazioni tecniche

Dove in principi si parla di come scegliere una mappa (prime due sezioni di @napo ) e nelle indicazioni tecniche si parla di librerie e dati.
Non saprei quanto deve essere approfondita per essere una linea guida :-)

**Comment by gunzip** [comment]
ho creato un nuovo branch:

https://github.com/italia-it/designer.italia.it/tree/55-standard-mappe

effettuato la relativa pull request e ho provveduto a commentarla:

https://github.com/italia-it/designer.italia.it/pull/144/files

necessita una revisione.

**Comment by sabas** [comment]
Il nuovo sito del comune di venezia usa Leaflet (bene)
però usa Esri per la cartografia / immagini satellitari.

Quando si svilupperà il template "standardizzato", probabilmente si potrebbe sviluppare un plugin Leaflet che fornisce accesso semplificato ai livelli italiani (geoportale nazionale, eccetera) sul modello di questi http://leafletjs.com/plugins.html#basemap-providers

**Comment by gunzip** [comment]
concordo che un esempio concerto da inserire nei template sarà efficace (è uno dei motivi per i quali questa issue è al momento procrastinata)
